File: MongoBooks2/BooksControllerUtilities/DataClasses/Author.cs
namespace BooksControllerUtilities.DataClasses
{
    using BooksCore.Books;

    public class Author
    {
        public string Name { get; set; }

        public string Nationality { get; set; }

        public string Language { get; set; }

        public int TotalPages { get; set; }

        public int TotalBooksReadBy { get; set; }
        
        public Book[] Books { get; set; }

        public Author()
        {

        }

        public Author(BookAuthor author)
        {
            Name = author.Author;
            Nationality = author.Nationality;
            Language = author.Language;
            TotalPages = author.TotalPages;
            TotalBooksReadBy = author.TotalBooksReadBy;
            Books = new Book[author.BooksReadBy.Count];
            for (int i = 0; i < author.BooksReadBy.Count; i++)
            {
                Books[i] = new Book( author.BooksReadBy[i] );
            }
        }
    }
}

File: MongoBooks2/HiGoogleCloud/ClientApp/src/app/Models/author.ts
import { Book } from './book'

export interface IAuthor
{
  name: string;
  nationality: string;
  language: string;
  totalPages: number;
  totalBooksReadBy: number;
  books: Book[];
};

export class Author implements IAuthor
{
  constructor(
    public name: string = "",
    public nationality: string = "",
    public language: string = "",
    public totalPages: number = 0,
    public totalBooksReadBy: number = 0,
    public books: Book[] = new Array<Book>())
  {
  }
}
